source/messaging/source/GreenEnergyHub.Messaging.Integration.ServiceCollection/HandlerExtensions.cs

```diff
@@ -82,5 +68,34 @@ public static void AddGreenEnergyHub(this IServiceCollection services, params Sy
 
             services.DiscoverValidation(assemblies);
         }
+
+        /// <summary>
+        /// Searches the provided <see cref="assemblies"/> for implementations of <see cref="interfaceToCheckFor"/>.
+        ///
+        /// Note, in case multiple types are found with the same Name, only one will be returned.
+        /// Which Type is selected, is based on the owning assembly's position in <see cref="assemblies"/>.
+        /// Types from assemblies with lower indexes are prioritized.
+        /// </summary>
+        /// <param name="interfaceToCheckFor">A type to use, as a basis, for finding the types to register</param>
+        /// <param name="assemblies">A sequence of assemblies, in which to search for implementations of <see cref="interfaceToCheckFor"/>.</param>
+        private static List<Type> FindImplementations(Type interfaceToCheckFor, System.Reflection.Assembly[] assemblies)
+        {
+            var hubMessageNameToTypeDictionary = new Dictionary<string, Type>();
+
+            // Check the assemblies in reverse order to satisfy the post condition that the first encounter of a given name wins.
+            // this works since '[type.Name] = type' adds/updates the type value stored, overriding values from less significant assemblies.
+            foreach (var assemblyToCheck in assemblies.Reverse())
+            {
+                var messageTypes = assemblyToCheck.GetTypes()
```

Comment:
The invocation of `.Reverse()`can be avoided by looping the array from last index to first

```suggestion
            for (var i = assemblies.Length - 1; i >= 0; i--)
            {
             var messageTypes = assemblies[i].GetTypes()
```

Reply:
Fair point, since i did argument from a performance perspective i should have done as you suggested.
My reasoning for not doing it, was readability - and given that a reverse is O(n) i did not mind the performance hit.

All taken into account - i think your suggestion is a better option